samza-log4j2/src/test/java/org/apache/samza/logging/log4j2/TestStreamAppender.java

```diff
@@ -444,6 +486,36 @@ private void logRecursivelyAndVerifyMessages(List<String> messages) throws Inter
     assertEquals(messages.size(), MockSystemProducer.messagesReceived.size());
   }
 
+  private void SetupSystemTimeoutAndVerifyMessages(long setUpSystemTime) throws InterruptedException {
+    MockSystemProducer.listeners.clear();
+    MockSystemAdmin.listeners.clear();
+    List<String> messages = Lists.newArrayList("testing1", "testing2");
+    // Set up latch
+    final CountDownLatch allMessagesSent = new CountDownLatch(messages.size());
+    MockSystemProducer.listeners.add((source, envelope) -> allMessagesSent.countDown());
+    MockSystemAdmin.listeners.add(streamSpec -> {
+      try {
+        // mock setUpSystem time during createStream() call
+        sleep(setUpSystemTime);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    });
+    ExecutorService service0 = Executors.newFixedThreadPool(1);
+    ExecutorService service1 = Executors.newFixedThreadPool(1);
+    // Log the messages with two threads
+    service0.submit(()->LOG.info(messages.get(0)));
+    service1.submit(()->LOG.info(messages.get(1)));
```

Comment:
Introduce a jitter so that there is no guarantees on the ordering of `setup`. Meaning `service1.submit(() -> LOG.info(messages.get(1)))` can also trigger `setup`

Reply:
Yes, service1 can also trigger setup. But the test doesn't care about ordering. The test only cares about the number of log events sent
If service1 holds the lock and sleeps for a while, then messages[1] will be sent and messages[0] will not. 
Validation remains the same

Reply:
precisely why I asked to add jitter to also cover the base that ordering doesn't matter as long as the messages emitted meets the criteria.

Reply:
oh ok, thanks for clarifying. will add it

Reply:
Added a random sleep time before LOG